Angular-Basic-Video-Chat/src/app/opentok.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+
+import * as OT from '@opentok/client';
+import config from '../config';
+
+@Injectable()
+export class OpentokService {
+
+  session: OT.Session;
+
+  constructor() { }
+
+  getOT() {
+    return OT;
+  }
+
+  connectSession() {
+    if (config.API_KEY && config.TOKEN && config.SESSION_ID) {
+      return this.connect(config.API_KEY, config.SESSION_ID, config.TOKEN);
+    } else {
+      return fetch(config.SAMPLE_SERVER_BASE_URL + '/session')
+        .then((data) => data.json())
+        .then((json) => this.connect(json.apiKey, json.sessionId, json.token));
+    }
```

Comment:
@aullman In case there's an error, can we have a  catch the error handle it appropriately?

Reply:
Yeah, I am handling it in the app.component.ts. It's not giving a very good error message though, I'll fix that.

Reply:
I noticed though we're not handling errors in the Basic Video Chat either. We should do that. Right now if you don't populate the config.js file you just get

```
Failed to load resource: net::ERR_NAME_NOT_RESOLVED
```

Reply:
@aullman will add that to the list! Thanks!